src/settings.cpp

```diff
@@ -558,6 +561,12 @@ void read_settings_xml(pugi::xml_node root)
     model::external_sources.push_back(make_unique<FileSource>(path));
   }
 
+  vector<double> source_strengths;
+  for (auto& s : model::external_sources) {
+    strengths.push_back(s->strength());
```

Comment:
This should be `source_strengths`
```suggestion
    source_strengths.push_back(s->strength());
```